cmd/server.go

```diff
@@ -42,129 +43,144 @@ var cacheSuffixes = []string{
 	".png",
 }
 
-var Serve = &cobra.Command{
-	Use:    "serve",
-	PreRun: PreRun,
-	Run: func(cmd *cobra.Command, args []string) {
-		e := echo.New()
-		e.HideBanner = true
-		e.Use(echoprometheus.NewMiddleware("mission_control"))
-		e.GET("/metrics", echoprometheus.NewHandler())
+func createHTTPServer(gormDB *gorm.DB) *echo.Echo {
+	e := echo.New()
+	e.HideBanner = true
 
-		// PostgREST needs to know how it is exposed to create the correct links
-		db.HttpEndpoint = publicEndpoint + "/db"
+	e.Use(func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			cc := api.NewContext(gormDB, c)
+			return next(cc)
+		}
+	})
+
+	e.Use(echoprometheus.NewMiddleware("mission_control"))
+	e.GET("/metrics", echoprometheus.NewHandler())
 
-		kratosHandler := auth.NewKratosHandler(kratosAPI, kratosAdminAPI, db.PostgRESTJWTSecret)
-		if enableAuth {
-			adminUserID, err := kratosHandler.CreateAdminUser(context.Background())
-			if err != nil {
-				logger.Fatalf("Failed to created admin user: %v", err)
-			}
-
-			middleware, err := kratosHandler.KratosMiddleware()
-			if err != nil {
-				logger.Fatalf("Failed to initialize kratos middleware: %v", err)
-			}
-			e.Use(middleware.Session)
-
-			// Initiate RBAC
-			if err := rbac.Init(adminUserID); err != nil {
-				logger.Fatalf("Failed to initialize rbac: %v", err)
-			}
+	kratosHandler := auth.NewKratosHandler(kratosAPI, kratosAdminAPI, db.PostgRESTJWTSecret)
+	if enableAuth {
+		adminUserID, err := kratosHandler.CreateAdminUser(context.Background())
+		if err != nil {
+			logger.Fatalf("Failed to created admin user: %v", err)
 		}
 
-		if !enableAuth {
-			db.PostgresDBAnonRole = "postgrest_api"
+		middleware, err := kratosHandler.KratosMiddleware()
+		if err != nil {
+			logger.Fatalf("Failed to initialize kratos middleware: %v", err)
 		}
-		if !disablePostgrest {
-			go db.StartPostgrest()
-			forward(e, "/db", "http://localhost:3000", rbac.Authorization(rbac.ObjectDatabase, "any"))
+		e.Use(middleware.Session)
+
+		// Initiate RBAC
+		if err := rbac.Init(adminUserID); err != nil {
+			logger.Fatalf("Failed to initialize rbac: %v", err)
 		}
+	}
 
-		if externalPostgrestUri != "" {
-			forward(e, "/db", externalPostgrestUri, rbac.Authorization(rbac.ObjectDatabase, "any"))
+	if !disablePostgrest {
+		forward(e, "/db", "http://localhost:3000", rbac.Authorization(rbac.ObjectDatabase, "any"))
+	} else if externalPostgrestUri != "" {
```

Comment:
This needs to be flipped around, if !disablePostgres && externalPostgresUri != "" else if !disablePostgres

Reply:
I see, so `externalPostgresUri` take precedence over the `disablePostgres` flag?

Reply:
instead of these flags, can we take the input of `--posrgres-uri` as
```
postgrest_opt = One of (none, local-bin, "uri-to-proxy")
```
and take action based on this ?

Reply:
@yashmehrotra I don't think the multi-valued value is more user friendly as there are different types for the value
@adityathebe disable-postgrest takes precedence, but the way the if statement evaluates:

disable == true && external == "" -> do nothing
disable == true && external != "" -> use external (but it shouldn't because its disabled)
!disable && external != "" -> use localhost (it should be using external)

Reply:
I guess that would work.

Reply:
> disable == true && external != "" -> use external (but it shouldn't because its disabled)

Ah, I see. So we have a different understanding of the `disablePostgrest` flag. 

I thought the flag simply decided whether we fire up our own postgrest server or not. Either we fireup a postgrest server or use an external one, but there has to be a Postgrest server. Without postgrest, would `incident-commander` even be functional?

Reply:
> Without postgrest, would incident-commander even be functional?
Yeah, there cases for it. For example, we might want to run it in a cluster in agent-mode just to get data on pods and nodes, in that scenario postgrest does not serve any function.

Postgrest is only required when we run the UI

Reply:
clarified with Yash. Will make that change.

Reply:
> `http://localhost:<port>` (to start postgrest on the user given port)

@yashmehrotra A slight problem is that I run my own postgrest instance locally and then pass that as an externalURL to incident commander. The server is running locally but I don't want incident commander to fire its own postgrest. wdyt?

I don't remember exactly why I am running standalone postgrest externally, but that has been my local setup since the beginning.

Reply:
umm, use `http://0.0.0.0:port` to pass that?

Reply:
That'd work for me but seems hacky to design it like that.

Reply:
Hows that hacky ?

Reply:
I think, it's natural to supply "localhost"

Reply:
The 0.0.0.0 suggestion is for your local to be working with this